### PR TITLE
Kpmp 312 normalize users

### PR DESCRIPTION
**Pull request by zwright**

### CODE REVIEW

File: src/main/java/org/kpmp/shibboleth/ShibbolethAttributeController.java [hunk 1]
```diff
@@ -19,23 +20,26 @@
...
+	private ShibbolethUserService shibbolethUserService;
 
 	@Autowired
-	public ShibbolethAttributeController(UTF8Encoder encoder) {
+	public ShibbolethAttributeController(UTF8Encoder encoder, ShibbolethUserService shibbolethUserService) {
 		this.encoder = encoder;
+		this.shibbolethUserService = shibbolethUserService;
 	}
 
 	// userMap(request)
```
Comment (rlreamy) [inline src/main/java/org/kpmp/shibboleth/ShibbolethAttributeController.java]: What are these comments about?  Can we remove them?

File: src/main/java/org/kpmp/shibboleth/UTF8Encoder.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import org.springframework.stereotype.Component;
 
 @Component
-class UTF8Encoder {
+public class UTF8Encoder {
```
Comment (rlreamy) [inline src/main/java/org/kpmp/shibboleth/UTF8Encoder.java]: Why did you make this public?  The only place it is getting used in the shibboleth package, and that is really the only place we want to use it.

File: src/main/java/org/kpmp/users/User.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package org.kpmp.users;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
```
Comment (rlreamy) [inline src/main/java/org/kpmp/users/User.java]: Looks like we are missing the json ordering annotation here.

File: src/test/java/org/kpmp/packages/PackageServiceTest.java [hunk 4]
```diff
@@ -40,6 +40,7 @@
 	private PackageZipService packageZipService;
 	@Mock
 	private FilePathHelper filePathHelper;
+	@Mock
```
Comment (rlreamy) [inline src/test/java/org/kpmp/packages/PackageServiceTest.java]: Pretty sure you don't want to mock the thing you are testing.

File: src/test/java/org/kpmp/packages/PackageTest.java [hunk 5]
```diff
@@ -155,8 +143,7 @@ public void testGenerateJSON() throws Exception {
 
 		assertEquals(
 				"{\"packageId\":\"packageId\",\"createdAt\":\"" + createdAtString + "\","
-						+ "\"packageType\":\"packageType\"," + "\"submitterFirstName\":\"submitterFirstName\","
-						+ "\"submitterLastName\":\"submitterLastName\"," + "\"submitterEmail\":\"submitter@email.com\","
+						+ "\"packageType\":\"packageType\"," + "\"submitter\":null,"
```
Comment (rlreamy) [inline src/test/java/org/kpmp/packages/PackageTest.java]: Let's fill out a user in the setup and make sure that the values come out in the expected order.  this test really captures the ordering desired in the metadata.json file.

File: src/main/java/org/kpmp/shibboleth/UTF8Encoder.java (hunk 2 above)
Comment (zwright) [reply src/main/java/org/kpmp/shibboleth/UTF8Encoder.java]: At one point I was going to use it outside the package then changed my mind.

**Pull request merged**